exposure-notification/src/test/java/fi/thl/covid19/exposurenotification/efgs/FederationServiceWithDaoTestIT.java

```diff
@@ -106,8 +140,8 @@ public void uploadKeys() {
                         .body("")
                 );
         dd.addKeys(1, md5DigestAsHex("test".getBytes()), to24HourInterval(Instant.now()), keyGenerator.someKeys(5), 5);
-        // TODO: check keys are bound to correct operation
```

Comment:
This verifies that the started operation id matches the one that was queued before. If you want to verify that the keys are bound to that operation, you could grab them from that keyGenerator.someKeys -call and run a separate sql after the insert to check that their operation id matches it also.

Reply:
Yep, this is true. However, keys and operations are cleared before test, so probably just checking that all keys have correct operation id would do. Dunno.

Reply:
Yeah, that's valid too